package proto_ff;

import "yd_fieldoptions.proto";


message monstermonster
{
	optional int64 monsterID = 1[(yd_fieldoptions.field_cname) = "怪物ID"];
	optional string monsterName = 2[(yd_fieldoptions.field_cname) = "怪物名", (yd_fieldoptions.field_bufsize) = 60];
	optional int32 functionType = 3[(yd_fieldoptions.field_cname) = "功能类型"];
	optional int64 functionValue = 4[(yd_fieldoptions.field_cname) = "功能参数"];
	optional int32 profession = 5[(yd_fieldoptions.field_cname) = "怪物职业"];
	optional int32 damageAddType = 6[(yd_fieldoptions.field_cname) = "伤害加成类型"];
	optional int32 monsterLV = 7[(yd_fieldoptions.field_cname) = "怪物等级"];
	optional int32 droplimit = 8[(yd_fieldoptions.field_cname) = "最大掉落等级"];
	optional int64 displayID = 9[(yd_fieldoptions.field_cname) = "资源与表现ID"];
	optional int64 valuetype = 10[(yd_fieldoptions.field_cname) = "数值类型"];
	optional int64 aiId = 11[(yd_fieldoptions.field_cname) = "AI表ID"];
	optional int32 hpSectionNum = 12[(yd_fieldoptions.field_cname) = "怪物血条段数"];
	optional int32 showGuild = 13[(yd_fieldoptions.field_cname) = "需要记录公会伤血"];
	optional int32 breedProb = 14[(yd_fieldoptions.field_cname) = "死亡触发新怪物几率"];
	optional int64 breedMonID = 15[(yd_fieldoptions.field_cname) = "触发生成怪物ID"];
	optional int32 breedMonLive = 16[(yd_fieldoptions.field_cname) = "怪物存活时间"];
	optional string itemCost = 17[(yd_fieldoptions.field_cname) = "道具消耗", (yd_fieldoptions.field_bufsize) = 60];
	optional int32 isBornActionOn = 18[(yd_fieldoptions.field_cname) = "是否启用出生动作"];
	optional int32 exp = 19[(yd_fieldoptions.field_cname) = "经验产出"];
	optional int32 dropType = 20[(yd_fieldoptions.field_cname) = "掉落类型"];
	optional int32 isBelongToPlayer = 21[(yd_fieldoptions.field_cname) = "怪物掉落归属是否显示"];
	optional int64 dropDefault = 22[(yd_fieldoptions.field_cname) = "默认掉落"];
	optional int64 dropBoxID = 23[(yd_fieldoptions.field_cname) = "掉落宝箱ID"];
	optional int32 ifMultiple = 24[(yd_fieldoptions.field_cname) = "单份多份掉落"];
	optional int64 dropRool = 25[(yd_fieldoptions.field_cname) = "Rool点掉落"];
	optional int64 lastBox = 26[(yd_fieldoptions.field_cname) = "最后一刀box奖励id"];
	optional int32 btnRange = 27[(yd_fieldoptions.field_cname) = "按钮范围"];
	optional int64 btnId = 28[(yd_fieldoptions.field_cname) = "按钮ID"];
}


message Sheet_monstermonster
{
	repeated monstermonster monstermonster_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message monsterdisplayskillDesc
{
	optional int64 id = 1[(yd_fieldoptions.field_cname) = "id"];
}

message monsterdisplay
{
	optional int64 displayID = 1[(yd_fieldoptions.field_cname) = "资源与表现ID"];
	optional int32 modelOffsetsX = 2[(yd_fieldoptions.field_cname) = "UI模型X坐标偏移"];
	optional int32 modelOffsetsY = 3[(yd_fieldoptions.field_cname) = "UI模型Y坐标偏移"];
	optional int32 modelScale = 4[(yd_fieldoptions.field_cname) = "UI模型缩放比例 "];
	optional int32 modelRadius = 5[(yd_fieldoptions.field_cname) = "怪物模型半径"];
	optional int32 pathRadius = 6[(yd_fieldoptions.field_cname) = "寻路半径"];
	optional int32 collisionRadius = 7[(yd_fieldoptions.field_cname) = "寻路半径"];
	optional int32 rSpeed = 8[(yd_fieldoptions.field_cname) = "跑步速度"];
	repeated monsterdisplayskillDesc skill = 9[(yd_fieldoptions.field_cname) = "技能", (yd_fieldoptions.field_arysize) = 7];
}


message Sheet_monsterdisplay
{
	repeated monsterdisplay monsterdisplay_List = 1[(yd_fieldoptions.field_arysize)=300];
}

message monstervalue
{
	optional int64 ID = 1[(yd_fieldoptions.field_cname) = "数值ID"];
	optional int32 type = 2[(yd_fieldoptions.field_cname) = "怪物类型"];
	optional int32 monsterLV = 3[(yd_fieldoptions.field_cname) = "怪物等级"];
	optional int64 maxHp = 4[(yd_fieldoptions.field_cname) = "生命"];
	optional int64 attack = 5[(yd_fieldoptions.field_cname) = "攻击"];
	optional int64 defense = 6[(yd_fieldoptions.field_cname) = "防御"];
	optional int64 pierce = 7[(yd_fieldoptions.field_cname) = "穿透"];
	optional int64 divineAtt = 8[(yd_fieldoptions.field_cname) = "神圣攻击"];
	optional int64 divineDef = 9[(yd_fieldoptions.field_cname) = "神圣防御"];
	optional int64 hitRate = 10[(yd_fieldoptions.field_cname) = "命中几率"];
	optional int64 dodgeRate = 11[(yd_fieldoptions.field_cname) = "闪避几率"];
	optional int64 critRate = 12[(yd_fieldoptions.field_cname) = "暴击几率"];
	optional int64 toughRate = 13[(yd_fieldoptions.field_cname) = "暴击抵抗"];
	optional int64 killRate = 14[(yd_fieldoptions.field_cname) = "必杀几率"];
	optional int64 parryRate = 15[(yd_fieldoptions.field_cname) = "必杀抵抗"];
	optional int64 critDam = 16[(yd_fieldoptions.field_cname) = "暴击伤害"];
	optional int64 critRedu = 17[(yd_fieldoptions.field_cname) = "暴击减免"];
	optional int64 killDam = 18[(yd_fieldoptions.field_cname) = "必杀伤害"];
	optional int64 killRedu = 19[(yd_fieldoptions.field_cname) = "必杀减免"];
	optional int64 attDam = 20[(yd_fieldoptions.field_cname) = "伤害提升"];
	optional int64 attRedu = 21[(yd_fieldoptions.field_cname) = "伤害减免"];
	optional int64 fightingCapacity = 22[(yd_fieldoptions.field_cname) = "战斗力"];
	optional int32 isFightPress = 23[(yd_fieldoptions.field_cname) = "是否战力压制"];
	optional int64 exp = 24[(yd_fieldoptions.field_cname) = "经验产出"];
}


message Sheet_monstervalue
{
	repeated monstervalue monstervalue_List = 1[(yd_fieldoptions.field_arysize)=43000];
}
